src/main/kotlin/com/exactpro/th2/codec/fixng/FixNgCodec.kt

```diff
@@ -546,10 +549,14 @@ class FixNgCodec(dictionary: IDictionaryStructure, settings: FixNgCodecSettings)
             isRequired = isRequired
         )
 
+        private fun getFirstTag(message: IMessageStructure): Int = message.fields.values.first().let {
```

Comment:
issue: this method doesn't consider group counter, for example
dictionary has structure: counterA [ counterB { fieldC } ]
the method returns fieldC instead of counterB for the message structure related to counterA

Reply:
Could you please add test for this case

Reply:
added encode and decode nested groups tests